### ISSUE TITLE
Add depth param support to e_sankey

### ISSUE THREAD

**Issue by artemklevtsov** [JohnCoene/echarts4r#89]
Helpful to control layout.

**Comment by JohnCoene** [comment]
Yes indeed, thanks again Artem. However I wanted to keep things simple for the sankey chart and thus the user is essentially asked to only pass edges, nodes are inferred from the latter. I am unsure what the best solution to this is, perhaps create two other functions that allow more control (`e_sankey_nodes` and `e_sankey_edges`)? 

Keen on your input.

**Comment by artemklevtsov** [comment]
According documentation `depth` relies to `data` array. So we should handle it (if exists) in the `.build_sankey_nodes`.

**Comment by artemklevtsov** [comment]
About `e_sankey_nodes` and `e_sankey_edges`. I think this is not a bad idea. Although, right now there is no need for this.

**Comment by JohnCoene** [comment]
Maybe @helgasoft's solution for the treemap could be applied here
